### PR TITLE
fix(dropdown): dynamic positioning of dropdown overlay

### PR DESCRIPTION
**Pull request by Jaynil1611**
Fixes: #1245 ,  #1344

Previous PR: #1284 
This PR is an enhancement on the previous PR which includes dynamic positioning of dropdown overlay depending on the height & width of the children rendered inside it.

### CODE REVIEW

**Comment by saurabhdaware** [comment]
#1344 will also get resolved with this right? it seems to be fixed on storybook

**Comment by Jaynil1611** [comment]
> #1344 will also get resolved with this right? it seems to be fixed on storybook

Yes, have added this issue in PR description

File: packages/blade/src/components/Dropdown/DropdownOverlay.web.tsx [hunk 1]
```diff
@@ -42,18 +42,20 @@ to {
...
       transform: translateY(${makeSize(props.theme.spacing[3])});
       opacity: 0;
       z-index: 99;
+      pointer-events: ${props.isOpen ? 'all' : 'none'};
     `,
 );
 
 type DropdownOverlayProps = {
-  children: React.ReactNode;
+  children: React.ReactElement;
```
Comment (saurabhdaware) [inline packages/blade/src/components/Dropdown/DropdownOverlay.web.tsx:58]: is this same as ReactNode 🤔

**Review by saurabhdaware — DISMISSED** [review]
Looks good to me. Big contributions! Thank you so much 🥳

File: packages/blade/src/components/Dropdown/docs/stories.ts [hunk 2]
```diff
@@ -539,8 +539,6 @@ const WithControlledSelectStory = `
             <ActionList>
               <ActionListItem title="Mumbai" value="mumbai" />
               <ActionListItem title="Bangalore" value="bangalore" />
-              <ActionListItem title="Pune" value="pune" />
-              <ActionListItem title="Chennai" value="chennai" />
```
Comment (saurabhdaware) [inline packages/blade/src/components/Dropdown/docs/stories.ts:543]: unrelated change?

File: packages/blade/src/components/Dropdown/DropdownOverlay.web.tsx (hunk 1 above)
Comment (Jaynil1611) [reply packages/blade/src/components/Dropdown/DropdownOverlay.web.tsx:58]: Nope, ReactNode is a superset that includes ReactElement. [More Info](https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement)

File: packages/blade/src/components/Dropdown/docs/stories.ts (hunk 2 above)
Comment (Jaynil1611) [reply packages/blade/src/components/Dropdown/docs/stories.ts:543]: @saurabhdaware There is not enough space to render all options which flips the position of the dropdown in controlled select [story](https://61c19ee8d3d282003ac1d81c-fanhpizsbk.chromatic.com/?path=/docs/components-dropdown-dropdown--page).
image

Comment (saurabhdaware) [reply packages/blade/src/components/Dropdown/docs/stories.ts:543]: oh got it. btw, when there is no enough space on top and bottom both, I think we should prefer opening it in bottom only right?

Comment (Jaynil1611) [reply packages/blade/src/components/Dropdown/docs/stories.ts:543]: Yeah, I have made the change. Please review once

**Review by saurabhdaware — APPROVED** [review]
